File: autoit-dlib-com/src/binding/correlation_tracker.cpp
#include "correlation_tracker.h"

using namespace std;

namespace dlib {
    // ----------------------------------------------------------------------------------------

    void start_track(
        correlation_tracker& tracker,
        cv::Mat& img,
        const drectangle& bounding_box
    )
    {
        AUTOIT_ASSERT_THROW(img.depth() == CV_8U, "Unsupported image type, must be 8bit gray or RGB image.");
        AUTOIT_ASSERT_THROW(img.channels() == 1 || img.channels() == 3, "Unsupported image type, must be 8bit gray or RGB image.");

        if (img.channels() == 1)
        {
            tracker.start_track(cv_image<unsigned char>(img), bounding_box);
        }
        else
        {
            tracker.start_track(cv_image<bgr_pixel>(img), bounding_box);
        }
    }

    double update(
        correlation_tracker& tracker,
        cv::Mat& img
    )
    {
        AUTOIT_ASSERT_THROW(img.depth() == CV_8U, "Unsupported image type, must be 8bit gray or RGB image.");
        AUTOIT_ASSERT_THROW(img.channels() == 1 || img.channels() == 3, "Unsupported image type, must be 8bit gray or RGB image.");

        if (img.channels() == 1)
        {
            return tracker.update(cv_image<unsigned char>(img));
        }

        return tracker.update(cv_image<bgr_pixel>(img));
    }

    double update(
        correlation_tracker& tracker,
        cv::Mat& img,
        const drectangle& bounding_box
    )
    {
        AUTOIT_ASSERT_THROW(img.depth() == CV_8U, "Unsupported image type, must be 8bit gray or RGB image.");
        AUTOIT_ASSERT_THROW(img.channels() == 1 || img.channels() == 3, "Unsupported image type, must be 8bit gray or RGB image.");

        if (img.channels() == 1)
        {
            return tracker.update(cv_image<unsigned char>(img), bounding_box);
        }

        return tracker.update(cv_image<bgr_pixel>(img), bounding_box);
    }

}

File: src/declarations/correlation_tracker.js
module.exports = [
	["class dlib.correlation_tracker", "", ["/Simple"], [], "", ""],

	["dlib.correlation_tracker.correlation_tracker", "", [], [], "", ""],

	["dlib.correlation_tracker.start_track", "void", ["/Call=dlib::start_track", "/Expr=*this->__self->get(), $0"], [
		["cv::Mat", "image", "", []],
		["drectangle", "bounding_box", "", []],
	], "", ""],

	["dlib.correlation_tracker.start_track", "void", ["/Call=dlib::start_track", "/Expr=*this->__self->get(), $0"], [
		["cv::Mat", "image", "", []],
		["rectangle", "bounding_box", "", ["/Cast=rectangle"]],
	], "", ""],

	["dlib.correlation_tracker.update", "double", ["/Call=dlib::update", "/Expr=*this->__self->get(), $0"], [
		["cv::Mat", "image", "", []],
	], "", ""],

	["dlib.correlation_tracker.update", "double", ["/Call=dlib::update", "/Expr=*this->__self->get(), $0"], [
		["cv::Mat", "image", "", []],
		["drectangle", "bounding_box", "", []],
	], "", ""],

	["dlib.correlation_tracker.update", "double", ["/Call=dlib::update", "/Expr=*this->__self->get(), $0"], [
		["cv::Mat", "image", "", []],
		["rectangle", "bounding_box", "", ["/Cast=rectangle"]],
	], "", ""],

	["dlib.correlation_tracker.get_position", "drectangle", [], [], "", ""],

];
